storage-mirror-firestore/stress_test/bin/driver.ts

```diff
@@ -711,6 +711,135 @@ const backfillCmd = new commander.Command("backfill")
+  .allowUnknownOption(false)
+  .description("delete tombstone records from firestore")
+  .option(
+    "--project <project_id>",
+    "project id which has the extension installed"
+  )
+  .option(
+    "--instance-id <extension_instance_id>",
+    "instance id of the extension (which can be found with `firebase ext:list`).",
+    "storage-mirror-firestore"
+  )
+  .action(async (command: commander.Command) => {
+    interface CommandWithOpts extends commander.Command {
+      project?: string;
+      instanceId?: string;
+    }
+    interface ExtensionInstance {
+      state: string;
+      config: {
+        params: {
+          ITEMS_TOMBSTONES_NAME: string;
+          PREFIXES_TOMBSTONES_NAME: string;
+          FIRESTORE_ROOT: string;
+        };
+      };
+    }
+
+    const options = command as CommandWithOpts;
+    const auth = new GoogleAuth({
+      scopes: ["https://www.googleapis.com/auth/firebase.readonly"],
+    });
+    const authClient = await auth.getClient();
+    const projectId = options.project || (await auth.getProjectId());
+
+    let instance: ExtensionInstance;
+    let url = `https://firebaseextensions.googleapis.com/v1beta/projects/${projectId}/instances/${options.instanceId}`;
+    try {
+      const response = await authClient.request({ url });
+      instance = response.data as ExtensionInstance;
+    } catch (e) {
+      if (e.code === 404) {
+        console.error(
+          `Extension instance '${options.instanceId}' on project '${projectId}' was not found.`
+        );
+        process.exit(1);
+      }
+      throw e;
+    }
+
+    if (instance.state !== "ACTIVE") {
+      console.error(
+        `Extension instance '${options.instanceId}' on project '${projectId}' is not active.`
+      );
+      return;
+    }
+
+    const prompt = `Are you sure you want to delete tombstone records from Firestore for the project '${projectId}' and instance '${options.instanceId}'?`;
+    if (!(await confirm(prompt))) {
+      process.exit(0);
+    }
+
+    console.log("\nScanning project for tombstone records...\n");
+
+    admin.initializeApp({
+      credential: admin.credential.applicationDefault(),
+      projectId: projectId,
+    });
+
+    let countOfTombstones = 0;
+    const writer = admin.firestore().bulkWriter();
+
+    const itemTombstonesCollectionGroup = admin
+      .firestore()
+      .collectionGroup(instance.config.params.ITEMS_TOMBSTONES_NAME);
+    const itemTombstonesPartitions =
+      itemTombstonesCollectionGroup.getPartitions(42);
+    for await (const partition of itemTombstonesPartitions) {
+      const partitionSnapshot = await partition.toQuery().get();
+      const documents = partitionSnapshot.docs;
+      for (const document of documents) {
+        // Only delete if the path matches the extensions configured Firestore root.
+        if (
+          document.exists &&
+          document.ref.path.startsWith(instance.config.params.FIRESTORE_ROOT)
```

Comment:
To make sure I understand: this script needs a collection group query + filter by  `FIRESTORE_ROOT` instead of just a query at `FIRESTORE_ROOT` because tombstones can occur at any path beneath `FIRESTORE_ROOT`, so a query wouldn't find all tombstones?

Reply:
Correct, they can be nested; e.g. 

![image](https://user-images.githubusercontent.com/5347038/125964458-2d298abf-6055-4619-92b1-04e4357e01d4.png)

Reply:
thanks!